include/mbedtls/compat-2.x.h

```diff
@@ -22,8 +22,6 @@
  *  limitations under the License.
  */
 
-#if ! defined(MBEDTLS_DEPRECATED_REMOVED)
```

Comment:
Ah, you removed it completely?
What I ment is that when MBEDTLS_DEPRECATED_REMOVED is not defined, then this header file should generate compilation error. Something like:
```
#if ! defined(MBEDTLS_DEPRECATED_REMOVED)
... code code code
#else
#error "Error notifying about the usage of depreacted feature"
#endif
```

Reply:
I copied, as @gilles-peskine-arm suggested, the idea of this file from the [development_2.x branch](https://github.com/ARMmbed/mbedtls/blob/development_2.x/include/mbedtls/compat-1.3.h) and there was not such thing like #error generation so I thought that you are asking for removal of the MBEDTLS_DEPRECATED_REMOVED altogether and so I did. Do you suggest that I should revert the deletion and introduce the #error generation here?

Reply:
Yes, indeed, it would make sense that including `compat-2.x.h` is an error when `MBEDTLS_DEPRECATED_REMOVED` is defined. Including `compat-1.3.h` in 2.x is a no-op when `MBEDTLS_DEPRECATED_REMOVED`, but I don't think this is right.

Reply:
I don't think this is critical though — we can deprecate things more over time.

Reply:
> Do you suggest that I should revert the deletion and introduce the #error generation here?

Yes, exactly. But as Gilles pointed out, it's not critical so I am approving this PR anyways.